.github/workflows/build-wasm_of_ocaml.yml

```diff
@@ -0,0 +1,228 @@
+name: Build wasm_of_ocaml
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  build:
+    env:
+      OPAMJOBS: 2
+      OPAMYES: true
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        ocaml-compiler:
+          - 4.14.x
+          - 5.00.x
+          - 5.01.x
+          - 5.02.x
+        include:
+          - os: macos-latest
+            ocaml-compiler: 5.02.x
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Set git to use LF
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+          git config --global core.ignorecase false
+
+      - name: Install node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+
+      - name: Restore cached binaryen
+        id: cache-binaryen
+        uses: actions/cache/restore@v4
+        with:
+          path: binaryen
+          key: ${{ runner.os }}-binaryen-version_118
+
+      - name: Checkout binaryen
+        if: steps.cache-binaryen.outputs.cache-hit != 'true'
+        uses: actions/checkout@v4
+        with:
+          repository: WebAssembly/binaryen
+          path: binaryen
+          submodules: true
+          ref: version_118
+
+      - name: Install ninja (Ubuntu)
+        if: matrix.os == 'ubuntu-latest' && steps.cache-binaryen.outputs.cache-hit != 'true'
+        run: sudo apt-get install ninja-build
+
+      - name: Install ninja (MacOS)
+        if: matrix.os == 'macos-latest' && steps.cache-binaryen.outputs.cache-hit != 'true'
+        run: brew install ninja
+
+      - name: Build binaryen
+        if: steps.cache-binaryen.outputs.cache-hit != 'true'
+        working-directory: ./binaryen
+        run: |
+          cmake -G Ninja .
+          ninja
+
+      - name: Cache binaryen
+        if: steps.cache-binaryen.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          path: binaryen
+          key: ${{ runner.os }}-binaryen-version_118
+
+      - name: Set binaryen's path
+        run: |
+          echo "$GITHUB_WORKSPACE/binaryen/bin" >> $GITHUB_PATH
+
+      - name: Restore cached OCaml
```

Comment:
Is this still needed with setup-ocaml v3 ?

Reply:
Removing this cache adds about 30s to the github workflow. I guess it's ok to remove it since it adds some complexity.